packages/maskbook/src/extension/options-page/DashboardDialogs/Wallet.tsx

```diff
@@ -159,19 +190,14 @@ export function DashboardWalletCreateDialog(props: WrappedDialogProps<object>) {
                     <>
                         <form>
                             <TextField
-                                helperText={
-                                    checkInputLengthExceed(name)
-                                        ? t('input_length_exceed_prompt', {
-                                              name: t('wallet_name').toLowerCase(),
-                                              length: WALLET_OR_PERSONA_NAME_MAX_LEN,
-                                          })
-                                        : undefined
-                                }
+                                helperText={errors.name?.type === zod.ZodIssueCode.too_big && errors?.name?.message}
                                 required
                                 autoFocus
+                                name="name"
                                 label={t('wallet_name')}
-                                value={name}
-                                onChange={(e) => setName(e.target.value)}
+                                inputRef={register}
+                                defaultValue={name.current}
+                                onChange={(e) => (name.current = e.target.value)}
```

Comment:
When the tab is swtiched each time, it will trigger the form unregister, resulting in the user previous input is cleared. So I created a ref to store the user's input and set it to the defaultValue of Input

Reply:
I did try that, look this [commit](https://github.com/DimensionDev/Maskbook/pull/2345/commits/fa42831d5c9cdbf8ccbcf89daf69c747a39c1f39).
I tried to use a different form field, But I don't think this approach is perfect either.
The reason for this problem is that this Input is uncontrolled，And the three form used a common button(Dialog footer) to submit,This makes them not easy to split between.

This does not happen with formik because formik manages them on the local state.

I think we need to make some choices, and the fact that react-hook-form introduces itself with better performance may have something to do with this implementation

Reply:
Ok, I got an [exmaple](https://codesandbox.io/s/headless-sound-wgetb) from the document, If needed, I can use it in our demos